Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -991,6 +1065,60 @@ namespace xr
                         inputSource.AimSpace.Pose.Orientation.W = location.pose.orientation.w;
                     }
                 }
+                
+                // Get joint data
+                if (sessionImpl.HandData.HandsInitialized)
+                {
+                    XrHandJointsLocateInfoEXT jointLocateInfo{XR_TYPE_HAND_JOINTS_LOCATE_INFO_EXT};
+                    jointLocateInfo.baseSpace = m_impl->sessionImpl.SceneSpace;
+                    jointLocateInfo.time = m_impl->displayTime;
+
+                    auto handInfo = sessionImpl.HandData.handsInfo[idx];
+                    XrCheck(sessionImpl.HmdImpl.Extensions->xrLocateHandJointsEXT(handInfo.handTracker, &jointLocateInfo, &handInfo.handJointData.locations));
+                    
+                    auto& inputSource = InputSources[idx];
+                    inputSource.JointsTrackedThisFrame = handInfo.handJointData.locations.isActive;
+                        
+                    // Set up the handJoints vector, skipping the palm joint as webXR doesn't support it
+                    // xrLocateHandJointsEXT will always return the full joint set (or an error), so jointCount should never change
+                    if (inputSource.HandJoints.size() != handInfo.handJointData.locations.jointCount - 1)
+                    {
+                        inputSource.HandJoints = std::vector<JointSpace>(handInfo.handJointData.locations.jointCount - 1);
+                        assert(inputSource.HandJoints.size() == handInfo.handJointData.locations.jointCount - 1);
+                    }
+                    
+                    // Set the joints to tracked, and populate the fields. Otherwise, set joints to untracked
+                    if (inputSource.JointsTrackedThisFrame)
+                    {
+                        constexpr XrSpaceLocationFlags RequiredFlags =
+                            XR_SPACE_LOCATION_POSITION_VALID_BIT |
+                            XR_SPACE_LOCATION_ORIENTATION_VALID_BIT |
+                            XR_SPACE_LOCATION_POSITION_TRACKED_BIT |
+                            XR_SPACE_LOCATION_ORIENTATION_TRACKED_BIT;
+
+                        for (uint32_t i = 0; i < handInfo.handJointData.locations.jointCount - 1; i++)
+                        {
+                            auto joint = handInfo.handJointData.jointLocations[i + 1];
```

Comment:
> + 1 [](start = 80, length = 4)

nit: for legibility it may help to declare this as a constant JOINT_OFFSET = 1 somewhere compared to just having an arbitrary increment here